### DISCUSSION TITLE
PAGView在TextureView的SizeChange事件中崩溃

### DISCUSSION THREAD

**Discussion by coolstar1204** [Tencent/libpag#2310] [General]
PAG的版本是com.tencent.tav:libpag:4.1.49
发生崩溃的Android版本：Android 14,level 34

`
SIGSEGV(SEGV_MAPERR):

#00 pc 0000000000038434 /vendor/lib64/mt6855/libsrv_um.so [arm64-v8a::bb2d997c55373906621164665c4ac393]

#01 pc 0000000000009ba8 /vendor/lib64/mt6855/libsrv_um.so (PVRSRVFreeDeviceMemMIW+40) [arm64-v8a::bb2d997c55373906621164665c4ac393]

#02 pc 0000000000023554 /vendor/lib64/mt6855/libIMGegl.so (KEGLUnbindImage+16) [arm64-v8a::a49f6ebbbd69885032e24c12a3e2eba0]
#03 pc 000000000000bbfc /vendor/lib64/mt6855/libIMGegl.so [arm64-v8a::a49f6ebbbd69885032e24c12a3e2eba0]

#04 pc 000000000000a878 /vendor/lib64/mt6855/libIMGegl.so (KEGLGetDrawableParameters+1104) [arm64-v8a::a49f6ebbbd69885032e24c12a3e2eba0]
#05 pc 0000000000017000 /vendor/lib64/mt6855/libIMGegl.so (IMGeglMakeCurrent+2184) [arm64-v8a::a49f6ebbbd69885032e24c12a3e2eba0]

#06 pc 0000000000003150 /system/lib64/libEGL.so (_ZN7android13egl_display_t11makeCurrentEPNS_13egl_context_tES2_PvS3_S3_S3_S3_S3_+176) [arm64-v8a::b7b66d46b081b8cba44bbb3780119527]

#07 pc 000000000000c4f8 /system/lib64/libEGL.so [arm64-v8a::b7b66d46b081b8cba44bbb3780119527]

#08 pc 00000000000f4188 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#09 pc 00000000000f8d50 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#10 pc 00000000000f2754 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#11 pc 0000000000091790 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so (_ZN3pag10PAGSurface17freeCacheInternalEv+104) [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#12 pc 00000000000916e8 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so (_ZN3pag10PAGSurface10updateSizeEv+76) [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#13 pc 00000000000e97b0 /data/app/~~BkA2XRSdIekHCzd9ok_GyA==/cn.kuwo.player-gX5D33zeFFyRCX4VQpUOIA==/lib/arm64/libpag.so (Java_org_libpag_PAGSurface_updateSize+68) [arm64-v8a::967a4a3fb6ed37243d201be59201740f]

#14 pc 0000000000155630 /apex/com.android.art/lib64/libart.so [arm64-v8a::41ac8dfed2977c7a168799976d23a7f8]

#15 pc 00000000003bb730 /apex/com.android.art/lib64/libart.so (_ZN3art3jit3Jit14AddCompileTaskEPNS_6ThreadEPNS_9ArtMethodENS_15CompilationKindEb+276) [arm64-v8a::41ac8dfed2977c7a168799976d23a7f8]

#16 pc 000000000067bf6c /system/framework/arm64/boot-framework.oat [arm64-v8a::b3d72aa6f5cc91676f2f6253ef6aa1f6]

org.libpag.PAGView.onSurfaceTextureSizeChanged(SourceFile:2)

android.view.TextureView.onSizeChanged(TextureView.java:447)

android.view.View.sizeChange(View.java:24718)`

**Comment by domchen** [comment]
到https://github.com/Tencent/libpag/releases 页面下载对应的符号表，把正确的堆栈还原出来，目前没有任何有效的信息，都是混淆的内容。另外你的版本也太老了，好几年前的修过的bug也不会继续维护了。建议先升级一下。
